environment/src/main/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionService.java

```diff
@@ -119,9 +121,19 @@ private CloudContext getCloudContext(EnvironmentDto environment) {
     }
 
     private List<CloudResource> getResourcesForDeletion(EnvironmentDto environment) {
+        List<CloudResource> resources = new ArrayList<>();
         Optional<CloudResource> desCloudResourceOptional = resourceRetriever.findByEnvironmentIdAndType(environment.getId(),
                 ResourceType.AZURE_DISK_ENCRYPTION_SET);
-        return desCloudResourceOptional.map(List::of).orElse(List.of());
+        Optional<CloudResource> rgCloudResourceOptional = resourceRetriever.findByResourceReferenceAndStatusAndType("-CDP_DES-",
+                CommonStatus.CREATED, ResourceType.AZURE_RESOURCE_GROUP);
```

Comment:
No, it has to be looked up just like the DES resource, i.e. using `findByEnvironmentIdAndType()`. The DES RG (for Multi-RG) is specific to a given environment, it is not a global resource.

Reply:
For multi-Rg, resource groups are created separately for datahub, freeIpa, datalake and des. So how would `findByEnvironmentIdAndType()` return the RG created specifically for DES?

Reply:
Those RG `CloudResource`s  (more precisely, the service-specific `Resource`s) are persisted in either freeipa or core service (the latter applicable to `CloudResource` belonging to a DL or DH stack). The `resourceRetriever` instance here is specific to the environment service, therefore it will not have visibility to the other RGs, so lookups will always succeed.

In general, the service-specific `Resource` representing a `CloudResource` is always associated with some resource ID, typically a CRN. In our case, this is the environment CRN. Later on, for `Resource`s representing the RG for a FreeIPA, DL or DH DES, those will be associated with the respective freeipa / datalake / datahub CRNs.